adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/dataSources/github/GitHubAuth.kt

```diff
@@ -0,0 +1,29 @@
+package net.adoptopenjdk.api.v3.dataSources.github
+
+import net.adoptopenjdk.api.v3.CredentialAccessor
+import net.adoptopenjdk.api.v3.SystemCredentialAccessor
+import org.slf4j.LoggerFactory
+
+class GitHubAuth(private val credentialAccessor: CredentialAccessor = SystemCredentialAccessor()) {
+
+    companion object {
+        @JvmStatic
+        private val LOGGER = LoggerFactory.getLogger(this::class.java)
+    }
+
+    private val tokenName = "GITHUB_TOKEN"
```

Comment:
I think I would move this to CredentialAccessor and make it the default argument